modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -941,7 +950,17 @@ struct Infer: public cv::detail::KernelTag {
                             // and redirect our data producers to this memory
                             // (A memory dialog comes to the picture again)
                             IE::Blob::Ptr this_blob = extractBlob(*ctx, i);
-                            req.SetBlob(ctx->uu.params.input_names[i], this_blob);
+                            // NB: In case importNetwork preprocessing must be
+                            // passed as SetBlob argument.
+                            if (ctx->uu.params.kind == cv::gapi::ie::detail::ParamDesc::Kind::Load) {
```

Comment:
Does this `if-else` deserve to be a function?

Infer, InferROI, InferList, InferList2 contain this.

Reply:
Good point, done